sw/device/lib/dif/dif_sram_ctrl.c

```diff
@@ -64,6 +64,39 @@ static uint32_t sram_ctrl_get_status(const dif_sram_ctrl_t *sram_ctrl) {
   return mmio_region_read32(sram_ctrl->base_addr, SRAM_CTRL_STATUS_REG_OFFSET);
 }
 
+dif_result_t dif_sram_ctrl_scramble(const dif_sram_ctrl_t *sram_ctrl) {
+  if (sram_ctrl == NULL) {
+    return kDifBadArg;
+  }
+
+  if (sram_ctrl_locked_ctrl(sram_ctrl)) {
+    return kDifLocked;
+  }
+
+  // Scramble memory addresses.
```

Comment:
This comment and the one below may be misleading, since:
1) both the payload of a 32bit as well as its address will be scrambled
2) the memory is technically always scrambled. by writing to `SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT` we are just requesting a new key to parameterize the scrambling functions for data and address. the block has a default scrambling key that it resets to after reset.

Hence, I would recommend to rephrase this to something like: `// Issue request for new scrambling key`

Reply:
Done.